Add unit tests for questions model and fixed NYC error

StewartWallace1115 opened the pull request:
Links
-----
- Task: https://www.notion.so/upchieve/Write-example-unit-test-for-server-repo-eb24852880714737bb089ce7581a7154


Description
-----------
Added unit test to the Question model. Add several edge cases for the getCategories function. This causes the test to fail. A new task needs to be opened to fix the Question model. NYC wasn't configured correctly show percentages, but that is fixed. Also, added HTML coverage reporting, which shows the line numbers that are not covered. 

Edit: Turns out CircleCI wouldn't let me submit tests that are known to fail. 

Developer self-review checklist
-------------------------------
- [x] Task's requirements have been fully addressed
- [x] Potentially confusing code has been explained with comments
- [x] Posted a link to this PR on the Notion task
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] There are no new spelling/grammar mistakes in the UI, code, or documentation
- [ ] Branch has been deployed to staging, and all edge cases have been manually tested
- [x] All new code complies with our ESLint standards

jkeat commented:
@StewartWallace1115 did you say last night that you were gonna comment out the failing unit tests? for now that sounds like a good solution to me 👍

StewartWallace1115 commented:
@jkeat I fixed the Question model and updated the unit test while I was at it. I was going to use a separate PR for the Question model, but Git was disagreeing with me.

jkeat commented:
@StewartWallace1115 What do you think about using a test structure like the client repo where they're all grouped under a  `tests/` directory at the root of the project? It would make the `require()` statements a bit trickier but I think it'd be really good from an organizational standpoint.
